File: client/webflix/src/app/model/basket-summary.ts
import {Movie} from './movie';

export class BasketSummary {
  movies: Movie[];
  total: number;

  static empty(): BasketSummary {
    let empty = new BasketSummary();
    empty.movies = [];
    empty.total = 0;
    return empty;
  }
}

File: server/webflix/src/main/java/com/instil/webflix/basket/model/BasketSummary.java
package com.instil.webflix.basket.model;

import java.math.BigDecimal;

import com.instil.webflix.movies.model.Movie;

public class BasketSummary {
	private Iterable<Movie> movies;
	private BigDecimal total;

	public BasketSummary(Iterable<Movie> movies, BigDecimal total) {
		this.movies = movies;
		this.total = total;
	}

	public Iterable<Movie> getMovies() {
		return movies;
	}
	
	public BigDecimal getTotal() {
		return total;
	}
}

